app/Http/Controllers/Web/DevicesController.php

```diff
@@ -18,29 +18,17 @@ class DevicesController extends Controller
 {
     private $deviceModel;
     private $rfDeviceModel;
-    private $userModel;
     private $messagePublisher;
 
-    public function __construct(Device $deviceModel, RFDevice $rfDeviceModel, User $userModel, MessagePublisher $messagePublisher)
```

Comment:
I do like removing the dependency on `User` in this class.  Maybe tricky in the end though.